corehq/apps/app_manager/static/app_manager/js/menu.js

```diff
@@ -12,9 +12,9 @@ hqDefine("app_manager/js/menu", function() {
         $('#langs select').change(function () {
             var lang = $(this).find('option:selected').attr('value'),
                 loc = window.location,
-                baseUrl = loc.origin + loc.pathname,
-                querystring = loc.search + (loc.search ? '&' : '?') + "lang=" + lang;
-            $(document).attr('location', baseUrl + querystring + loc.hash);
+                params = $.unparam(loc.search.slice(1));
```

Comment:
Maybe add in a `if (loc.search)` somewhere? `$.unparam` will return `{"": true}` if given a blank string which is a bit funky.